Identity/Crypto.cs

```diff
@@ -74,6 +88,22 @@ public static bool VerifySignDetached(byte[] message, byte[] signature, byte[] p
                 var ok = Sodium.PublicKeyAuth.VerifyDetached(signature, message, publicKey);
                 return ok;
             }
+
+            public static bool ByteArrayCompare(byte[] left, byte[] right)
+            {
+                if (left.Length != right.Length)
+                {
+                    return false;
+                }
+                for (int i = 0; i < left.Length; i++)
+                {
+                    if (!left[i].Equals(right[i]))
+                    {
+                        return false;
+                    }
+                }
+                return true;
```

Comment:
There is a simpler approach (left.equal(right()). Which is slow, I know, but I don't think we should care. The proper way is to use the new `ReadOnlySpan<T>`, but we need some trickery to suppor every dotnet platforms. I would stick to the the boring left.Equal(right);

Reply:
hum... `byte[]` does not implement `equal`, that's why I did this stupid implementation.
Or I missed something?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.byte.equals?view=netcore-2.2

its `Equals()`, my bad. `Array.Equals()` works too.

Reply:
That's for a single byte struct, and that's what I did, right?